File: Caesar cipher/Caesarcipher.cpp
#include <iostream>
#include <string>

void cap_let (std::string&);
void caesar_cipher (std::string& );


int main ()
{
    std::string cipher = "Caesar cipher";
    cap_let (cipher);
    caesar_cipher (cipher);

    std::cout << cipher << std::endl;
}

void cap_let (std::string& str)
{
    int size = str.size ();
    for (int i = 0; i < size; ++i)
    {
        if (str[i] > 'Z') { str[i] -= 32; }
    }
}

void caesar_cipher (std::string& str) 
{
    const int alpsize = 26;
    int shift_count;
    std::cin >> shift_count;
    shift_count %=  alpsize;
    int size = str.size ();

        for (int i = 0; i < size; ++i)
    {
        if (str[i] == ' ') { continue ;}
        if (str[i] - shift_count < 'A')
        {
            str[i] += alpsize;
        }
        str[i] -= shift_count;
    }
}

File: Caesar cipher/Caesarcipher.js
function cap_let (str)
{
    let nstr = "";
    int size = str.length;
    
    for (let i = 0; i < size; i++)
    {
        if (str[i] > 'Z')
        {
            nstr += (String.fromCharCode (str.charCodeAt (i) - 32));
        }
        else
        {
            nstr += str[i];
        }
        
    }
    return nstr;
}

function caesar_cipher (str)
{
    const alpsize = 26;
    let shift_count = 3;
    let size = str.length;
    let nstr = "";
    
    for (let i = 0; i < size; i++)
    {
        if (str[i] === ' ') { continue; }
        if ((String.fromCharCode (str.charCodeAt (i) - shift_count)) < 'A') 
        {
            nstr += (String.fromCharCode (str.charCodeAt (i) + alpsize - shift_count));
        }
        else
        {
            nstr += (String.fromCharCode (str.charCodeAt (i) - shift_count));
        }
    }
    return nstr;
    
}

// MAIN

let str = "Caesar cipher";
let nstr = cap_let (str);
console.log (caesar_cipher (nstr));


